### PR TITLE
Add user audit logs page to dashboard

### PR DESCRIPTION
**Pull request by dsdshcym**
As explained in https://github.com/hexpm/hexpm/issues/754#issuecomment-571135459, this PR adds paginated audit logs table to dashboard:

image

As a bonus, this PR also extracts a `_pagination` partial, which renders pagination nav bar in both packages index page and /dashboard/audit_logs index page.

### CODE REVIEW

File: lib/hexpm_web/templates/dashboard/audit_log/index.html.eex [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+          <%= for audit_log <- @audit_logs do %>
+            <tr>
+              <td class="col-sm-3"><%= pretty_date(audit_log.inserted_at, :short) %></td>
+              <td><%= humanize_audit_log_info(audit_log) %></td>
+            </tr>
+          <% end %>
+        </table>
+
+        <%=
+        render HexpmWeb.LayoutView,
```
Comment (dsdshcym) [inline lib/hexpm_web/templates/dashboard/audit_log/index.html.eex]: Do we have a `SharedView` or something alike?
`LayoutView` doesn't seem to be the right place for this shared partial.

Maybe we can create a `SharedView` if needed?

File: lib/hexpm_web/templates/dashboard/audit_log/index.html.eex [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+               "_pagination.html",
```
Comment (dsdshcym) [inline lib/hexpm_web/templates/dashboard/audit_log/index.html.eex]: Any recommendations for a better template name?

`_pagination` feels too vague. 
`_pagination_nav` feels too narrow.

File: lib/hexpm_web/views/dashboard/audit_log_view.ex [hunk 3]
```diff
@@ -0,0 +1,100 @@
+defmodule HexpmWeb.Dashboard.AuditLogView do
+  use HexpmWeb, :view
+
+  alias HexpmWeb.DashboardView
+
+  @doc """
+  Translate an audit_log to user readable descriptions
+  """
+  def humanize_audit_log_info(log = %{action: "docs.publish"}) do
+    "Publish documentation for #{log.params["package"]["name"]} (#{
```
Comment (dsdshcym) [inline lib/hexpm_web/views/dashboard/audit_log_view.ex]: Do we need to handle the case when `params["package"]["name"]` is `nil`?

It won't crash now, but the humanized result would be weird: `Publish documentation for  ()`

File: lib/hexpm_web/controllers/dashboard/audit_log_controller.ex [hunk 4]
```diff
@@ -0,0 +1,24 @@
+defmodule HexpmWeb.Dashboard.AuditLogController do
+  use HexpmWeb, :controller
+
+  plug :requires_login
+
+  @per_page 10
```
Comment (ericmj) [inline lib/hexpm_web/controllers/dashboard/audit_log_controller.ex]: These queries are cheap so let's make this a large number, like `@per_page 100`.

File: lib/hexpm_web/templates/dashboard/audit_log/index.html.eex (hunk 1 above)
Comment (ericmj) [reply lib/hexpm_web/templates/dashboard/audit_log/index.html.eex]: Create `SharedView` if needed.

File: lib/hexpm_web/templates/dashboard/audit_log/index.html.eex (hunk 2 above)
Comment (ericmj) [reply lib/hexpm_web/templates/dashboard/audit_log/index.html.eex]: `_pagination` is fine imo.

Comment (ericmj) [reply lib/hexpm_web/templates/dashboard/audit_log/index.html.eex]: Good job extracting it btw.

File: lib/hexpm_web/views/dashboard/audit_log_view.ex (hunk 3 above)
Comment (ericmj) [reply lib/hexpm_web/views/dashboard/audit_log_view.ex]: We should always have a name in this case.

File: lib/hexpm_web/views/dashboard/audit_log_view.ex [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+  end
+
+  def humanize_audit_log_info(log = %{action: "owner.remove"}) do
+    "Remove #{log.params["user"]["username"]} from owners of package #{
+      log.params["package"]["name"]
+    }"
+  end
+
+  def humanize_audit_log_info(log = %{action: "release.publish"}) do
+    "Publish package #{log.params["package"]["name"]} version #{log.params["release"]["version"]}"
```
Comment (ericmj) [inline lib/hexpm_web/views/dashboard/audit_log_view.ex]: Sometimes we have `package version 1.0.0` and sometimes `package (1.0.0)`, we should be consistent.

Comment (dsdshcym) [reply lib/hexpm_web/views/dashboard/audit_log_view.ex]: updated in https://github.com/hexpm/hexpm/pull/886/commits/8a4ef6e9feccec8be98a0788fc7ebf0fb0dce3d3

File: lib/hexpm_web/controllers/dashboard/audit_log_controller.ex (hunk 4 above)
Comment (dsdshcym) [reply lib/hexpm_web/controllers/dashboard/audit_log_controller.ex]: updated: https://github.com/hexpm/hexpm/pull/886/commits/e89dfd49bbe15cd34228bf56d6eb6b685d3dd8ed

File: lib/hexpm_web/templates/dashboard/audit_log/index.html.eex (hunk 1 above)
Comment (dsdshcym) [reply lib/hexpm_web/templates/dashboard/audit_log/index.html.eex]: ok
https://github.com/hexpm/hexpm/pull/886/commits/b050ce51143d0b91bff55d6b284f392e25f88375

File: lib/hexpm_web/views/dashboard/audit_log_view.ex (hunk 3 above)
Comment (dsdshcym) [reply lib/hexpm_web/views/dashboard/audit_log_view.ex]: Then I'd suggest we keep the current simple logic and fix any issues as they occur, sounds good to you?

**Comment by ericmj** [comment]
Looks great! There is a conflict because we moved `ViewHelpers` from an import to an alias in https://github.com/hexpm/hexpm/commit/4c4f07397fc1c60a2c7647e57e40810ea7efa51d. So places like this https://github.com/hexpm/hexpm/pull/886/files#diff-da93f12215c3e361b100fcf30b095414R19 and this https://github.com/hexpm/hexpm/pull/886/files#diff-695f50dd681f76fb1eefcfd278909df9R27needs to change to use fully qualified calls: `ViewHelpers.function(...)`.

**Comment by dsdshcym** [comment]
Got it, updated in https://github.com/hexpm/hexpm/pull/886/commits/3f3d196faa40d177167cabeaa719bf9220427c95

**Comment by ericmj** [comment]
💜

**Pull request merged**